File: ToDo/ToDo.Web/Modules/Administration/MovieCast/MovieCastColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace ToDo.Administration.Columns;

[ColumnsScript("Administration.MovieCast")]
[BasedOnRow(typeof(MovieCastRow), CheckNames = true)]
public class MovieCastColumns
{
    //[EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    //public int MovieCastId { get; set; }
    //public string MovieTitle { get; set; }
    //public string PersonFullName { get; set; }
    //[EditLink]
    //public string Character { get; set; }

    [EditLink, Width(250)]
    public string PersonFullName { get; set; }
    [EditLink, Width(250)]
    public string Character { get; set; }
}

File: ToDo/ToDo.Web/Modules/ServerTypes/Administration/MovieCastColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { MovieCastRow } from "./MovieCastRow";

export interface MovieCastColumns {
    PersonFullName: Column<MovieCastRow>;
    Character: Column<MovieCastRow>;
}

export class MovieCastColumns extends ColumnsBase<MovieCastRow> {
    static readonly columnsKey = 'Administration.MovieCast';
    static readonly Fields = fieldsProxy<MovieCastColumns>();
}